File: look_flutter/lib/entity/response/video_response.dart
class VideoResponse {
  List<VideoEntity> records;
  int total;
  int size;
  int current;
  bool optimizeCountSql;
  bool searchCount;
  int pages;

  VideoResponse(
      {this.records,
        this.total,
        this.size,
        this.current,
        this.optimizeCountSql,
        this.searchCount,
        this.pages});

  VideoResponse.fromJson(Map<String, dynamic> json) {
    if (json['records'] != null) {
      records = <VideoEntity>[];
      json['records'].forEach((v) {
        records.add(VideoEntity.fromJson(v));
      });
    }
    total = json['total'];
    size = json['size'];
    current = json['current'];
    optimizeCountSql = json['optimizeCountSql'];
    searchCount = json['searchCount'];
    pages = json['pages'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    if (records != null) {
      data['records'] = records.map((v) => v.toJson()).toList();
    }
    data['total'] = total;
    data['size'] = size;
    data['current'] = current;
    data['optimizeCountSql'] = optimizeCountSql;
    data['searchCount'] = searchCount;
    data['pages'] = pages;
    return data;
  }
}

class VideoEntity {
  String id;
  String memberId;
  String title;
  String subTitle;
  String tagNameList;
  String cover;
  String videoUrl;
  num seconds;
  int width;
  int height;
  bool watchType;
  String canWatchMember;
  String cannotWatchMember;
  String channel;
  String address;
  num longitude;
  num latitude;
  String createdAt;
  String avatar;
  String nickname;
  num lickCount;
  num commentCount;
  num shareCount;
  int hasLiked;

  VideoEntity(
      {this.id,
        this.memberId,
        this.title,
        this.subTitle,
        this.tagNameList,
        this.cover,
        this.videoUrl,
        this.seconds,
        this.width,
        this.height,
        this.watchType,
        this.canWatchMember,
        this.cannotWatchMember,
        this.channel,
        this.address,
        this.longitude,
        this.latitude,
        this.createdAt,
        this.avatar,
        this.nickname,
        this.lickCount,
        this.commentCount,
        this.shareCount,
        this.hasLiked});

  VideoEntity.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    memberId = json['memberId'];
    title = json['title'];
    subTitle = json['subTitle'];
    tagNameList = json['tagNameList'];
    cover = json['cover'];
    videoUrl = json['videoUrl'];
    seconds = json['seconds'];
    width = json['width'];
    height = json['height'];
    watchType = json['watchType'];
    canWatchMember = json['canWatchMember'];
    cannotWatchMember = json['cannotWatchMember'];
    channel = json['channel'];
    address = json['address'];
    longitude = json['longitude'];
    latitude = json['latitude'];
    createdAt = json['createdAt'];
    avatar = json['avatar'];
    nickname = json['nickname'];
    lickCount = json['lickCount'];
    commentCount = json['commentCount'];
    shareCount = json['shareCount'];
    hasLiked = json['hasLiked'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['memberId'] = memberId;
    data['title'] = title;
    data['subTitle'] = subTitle;
    data['tagNameList'] = tagNameList;
    data['cover'] = cover;
    data['videoUrl'] = videoUrl;
    data['seconds'] = seconds;
    data['width'] = width;
    data['height'] = height;
    data['watchType'] = watchType;
    data['canWatchMember'] = canWatchMember;
    data['cannotWatchMember'] = cannotWatchMember;
    data['channel'] = channel;
    data['address'] = address;
    data['longitude'] = longitude;
    data['latitude'] = latitude;
    data['createdAt'] = createdAt;
    data['avatar'] = avatar;
    data['nickname'] = nickname;
    data['lickCount'] = lickCount;
    data['commentCount'] = commentCount;
    data['shareCount'] = shareCount;
    data['hasLiked'] = hasLiked;
    return data;
  }
}

File: look-server/look-video/look-video-web/src/main/java/com/whoiszxl/cqrs/response/VideoResponse.java
package com.whoiszxl.cqrs.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * <p>
 * 视频表
 * </p>
 *
 * @author whoiszxl
 * @since 2022-05-06
 */
@Getter
@Setter
@ApiModel(value = "Video对象", description = "视频表")
public class VideoResponse implements Serializable {

    private static final long serialVersionUID = 1L;

    @ApiModelProperty("视频主键ID")
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private Long id;

    @ApiModelProperty("会员ID")
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private Long memberId;

    @ApiModelProperty("标题")
    private String title;

    @ApiModelProperty("副标题")
    private String subTitle;

    @ApiModelProperty("标签名列表")
    private String tagNameList;

    @ApiModelProperty("视频封面")
    private String cover;

    @ApiModelProperty("视频文件地址")
    private String videoUrl;

    @ApiModelProperty("视频秒数")
    private Integer seconds;

    @ApiModelProperty("视频宽度")
    private Integer width;

    @ApiModelProperty("视频高度")
    private Integer height;

    @ApiModelProperty("观看类型:(1:所有人 2:私密 3:好友可见 4:部分可见 5:对谁不可见)")
    private Boolean watchType;

    @ApiModelProperty("可见用户的id列表,使用逗号 , 分隔")
    private String canWatchMember;

    @ApiModelProperty("不可见用户的id列表,使用逗号 , 分隔")
    private String cannotWatchMember;

    @ApiModelProperty("来自什么渠道")
    private String channel;

    @ApiModelProperty("定位地址")
    private String address;

    @ApiModelProperty("经度")
    private BigDecimal longitude;

    @ApiModelProperty("纬度")
    private BigDecimal latitude;

    @ApiModelProperty("创建时间")
    private LocalDateTime createdAt;


    @ApiModelProperty("头像")
    private String avatar;

    @ApiModelProperty("昵称")
    private String nickname;

    @ApiModelProperty("点赞数")
    private Integer lickCount;

    @ApiModelProperty("评论数")
    private Integer commentCount;

    @ApiModelProperty("分享数")
    private Integer shareCount;

    @ApiModelProperty("是否点赞 1:是 0:否")
    private Integer hasLiked;

}
